app/src/main/java/com/cmput301f20t21/bookfriends/ui/browse/BrowseDetailViewModel.java

```diff
@@ -20,13 +24,22 @@ public BrowseDetailViewModel() {
 
     public BrowseDetailViewModel(AuthRepository authRepository, RequestRepository requestRepository) {
         this.requestRepository = requestRepository;
+        this.authRepository = authRepository;
         currentUsername = authRepository.getCurrentUser().getUsername();
     }
 
     public void sendRequest(Book book, OnSuccessCallback successCallback, OnFailCallback failCallback) {
         requestRepository
                 .sendRequest(currentUsername, book.getId())
-                .addOnSuccessListener(requestId -> successCallback.run())
+                .addOnSuccessListener(requestId -> {
+                    NotificationSender.getInstance().request(book, authRepository.getCurrentUser(), res -> {
```

Comment:
for the User class, the only action we are doing is to get the username in .request(), we already have `currentUsername` in the class attribute, can we just pass that instead?